src/main/java/us/abstracta/wiresham/FlowFactory.java

```diff
@@ -0,0 +1,187 @@
+      String targetAddress) {
+    this.configFile = configFile;
+    this.serverAddress = serverAddress;
+    this.pcapFilter = pcapFilter;
+    this.port = targetAddress == null ? port : getPortFromTargetAddress(targetAddress);
+    this.hostTarget = getHostFromTargetAddress(targetAddress);
+  }
+
+  private String getHostFromTargetAddress(String targetAddress) {
+    if (targetAddress == null) {
+      return null;
+    }
+    int hostPortSeparator = targetAddress.lastIndexOf(":");
+    return targetAddress.substring(0, hostPortSeparator);
+  }
+
+  private int getPortFromTargetAddress(String targetAddress) {
+    int portSeparatorPos = targetAddress.lastIndexOf(":");
+    return Integer.parseInt(targetAddress.substring(portSeparatorPos + 1));
+  }
+
+  public List<Flow> getFlows() {
+    List<Flow> flows;
+    try {
+      if (serverAddress != null) {
+        flows = Collections.singletonList(buildWiresharkFlow());
+      } else {
+        flows = buildFromYamlFile();
+      }
+
+      if (flows.size() == 1 && flows.get(0).getPort() == null) {
+        flows.get(0).setPort(port);
+      }
+      return flows;
+    } catch (IOException e) {
+      LOG.error("There was an error trying to read the wireshark flow", e);
+      System.err.println(e.getMessage());
+      return null;
+    }
+  }
+
+  private List<Flow> buildFromYamlFile() throws FileNotFoundException {
+    List<PacketStep> packets = new Yaml(buildYamlConstructor())
+        .load(new FileInputStream(configFile));
+    return dissectStepsIntoFlow(packets);
+  }
+
+  private static Constructor buildYamlConstructor() {
+    Constructor constructor = new Constructor();
+    YAML_TAGS
+        .forEach((tag, clazz) -> constructor.addTypeDescription(new TypeDescription(clazz, tag)));
+    return constructor;
+  }
+
+  public List<Flow> dissectStepsIntoFlow(List<PacketStep> packetSteps) {
+    List<List<PacketStep>> dissectedFlow = new ArrayList<>();
+    List<PacketStep> currentFlowPort = new ArrayList<>();
+    for (PacketStep step : packetSteps) {
+      if ((step instanceof SendPacketStep) && ((SendPacketStep) step).getPort() != null) {
+        if (!currentFlowPort.isEmpty()) {
+          dissectedFlow.add(currentFlowPort);
+        }
+        currentFlowPort = new ArrayList<>();
+      }
+      currentFlowPort.add(step);
+    }
+    dissectedFlow.add(currentFlowPort);
+    return buildFlowFromDissectedPackets(dissectedFlow);
+  }
+
+  private Flow buildWiresharkFlow() throws IOException {
+    if (configFile.getName().toLowerCase().endsWith(".json")) {
+      return Flow.fromWiresharkJsonDump(configFile, serverAddress);
+    } else {
+      return Flow.fromPcap(configFile, serverAddress, pcapFilter);
+    }
+  }
+
+  public void saveYml(File ymlFile) throws IOException {
+    new Yaml(buildYamlRepresenter())
+        .dump(getFlows(), new FileWriter(ymlFile));
+  }
+
+  private static Representer buildYamlRepresenter() {
+    Representer representer = new Representer() {
```

Comment:
does it make sense to put a comment why this representer is needed? it seems that converts 0 longs to null (whatever the long is) or 0 int ports to null.